core/Services/ReportService.cs

```diff
@@ -4,19 +4,21 @@
 using api.core.Data.Responses;
 using api.core.repositories.abstractions;
 using api.core.services.abstractions;
+using api.emails.Models;
+using api.emails.Services.Abstractions;
 
 
 namespace api.core.Services;
 
-public class ReportService(IEventRepository eventRepository, IReportRepository reportRepository) : IReportService
+public class ReportService(IEventRepository eventRepository, IEventService eventService, IEmailService emailService, IReportRepository reportRepository) : IReportService
 {
     public IEnumerable<ReportResponseDTO> GetReports()
     {
         var reports = reportRepository.GetAll();
         return reports.Select(ReportResponseDTO.Map).ToList();
     }
 
-    public void ReportEvent(Guid eventId, CreateReportRequestDTO request)
+    public async void ReportEvent(Guid eventId, CreateReportRequestDTO request)
```

Comment:
Faut éviter les async void en c# -> https://stackoverflow.com/questions/12144077/async-await-when-to-return-a-task-vs-void

T'es supposé modifié ton retour avec un `Task` et comme bonne pratique de rajouter un Async à la fin (exemple: `MethodNameAsync`)

Reply:
Fixed